go/consensus/cometbft/apps/keymanager/churp/txs.go

```diff
@@ -545,3 +556,20 @@ func resetHandoff(status *churp.Status, nextHandoff beacon.EpochTime) {
 	status.NextChecksum = nil
 	status.Applications = nil
 }
+
+func verifyPolicy(ctx *tmapi.Context, policy *churp.SignedPolicySGX) error {
+	// Allow non-empty `MayQuery` field with the 24.2 release.
+	regState := registryState.NewMutableState(ctx.State())
+	regParams, err := regState.ConsensusParameters(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to load registry consensus parameters: %w", err)
+	}
+	if regParams.SoftwareVersion != nil && regParams.SoftwareVersion.ToU64() >= version.MustFromString("24.2").ToU64() {
```

Comment:
Can we have some sort of helper for this? So we can do something like `IsFeatureVersion(ctx, "24.2")` and it would do all the required things? Also rename the field to something like `FeatureVersion` and move to root consensus parameters.